app/controllers/api/api_base_controller.rb

```diff
@@ -98,6 +98,19 @@ def current_account
       current_user&.account
     end
 
+    def current_ability
+      @current_ability ||= Ability.new(current_user, partnership_request_context)
+    end
+
+    def partnership_request_context
+      return nil if params[:accessible_partnership_ids].blank?
+
+      {
+        accessible_partnership_ids: Array.wrap(params[:accessible_partnership_ids]),
+        external_account_id: params[:external_account_id]
+      }
+    end
```

Comment:
We don't store partnership relationships in the database. So the API call provides to Docuseal what partnerships the user has access to, so this first needs to be provided to CanCan for authorization.